### PR TITLE
Deterministic sorting ++

### PR DESCRIPTION
**Pull request by Velenir**
- Simpler `_fetchToken`
- Sorting in one pass and without splicing
- Unicode-aware sorting for possible crazy symbols and just ease of use
- Emphasis on mutable sorting, so we don't have wrong expectations

### CODE REVIEW

**Comment by gnosis-info** [comment]
Travis automatic deployment:
   * **🎩 [Dapp](https://pr1517--dexreact.review.gnosisdev.com)**: Testing web app
   * **📚 [Storybook](https://pr1517--dexreact.review.gnosisdev.com/storybook/)**: Component stories

**Review by anxolin — APPROVED** [review]
Nice, good stuff

File: src/services/factories/tokenList.ts [hunk 1]
```diff
@@ -173,57 +173,62 @@ export function getTokensFactory(factoryParams: {
...
+          // WETH first
+          if (a.address === WETH_ADDRESS_RINKEBY) return -1
+          return compareByLabel(a, b)
+        }
       case Network.xDAI:
-        tokensSorted = _moveTokenToHeadOfArray(WETH_ADDRESS_XDAI, tokensSorted)
-        return _moveTokenToHeadOfArray(WXDAI_ADDRESS_XDAI, tokensSorted)
+        comparator = (a, b): number => {
+          // WXDAI before WETH
+          if (a.address === WXDAI_ADDRESS_XDAI && b.address === WETH_ADDRESS_XDAI) return -1
```
Comment (anxolin) [inline src/services/factories/tokenList.ts:219]: this part get's a bit complicated

**Review by anxolin — COMMENTED** [review]
Please dima, note that with the current sorting, WETH and wxDAI are placed at the bottom

**Review by anxolin — COMMENTED** [review]
image

**Review by W3stside — APPROVED** [review]

**Pull request merged**